File: src/modules/text.ts
import Base from "../base.js";
import { EventEmitter } from "events";

export default class Text extends Base {
  constructor(start: {
    apiKey: string;
    captchaKey: string;
    options?: {
      stream?: boolean;
      host?: string;
    };
  }) {
    super(start);
  }
  async alan(data: {
    userName: string;
    messages: any[];
    searchEngine?: string;
    image?: string;
    imageDescription?: string;
    imageGenerator?: string;
    nsfw?: boolean;
    audioGenerator?: string;
    imageModificator?: string;
    max_tokens?: number;
  }): Promise<EventEmitter | any> {
    return await this.fetch(`${this.options.host}/text/alan`, data);
  }
  async anthropic(data: {
    messages: any[];
    model?: string;
    max_tokens?: number;
    temperature?: number;
    stream?: boolean;
  }): Promise<EventEmitter | any> {
    return await this.fetch(`${this.options.host}/text/anthropic`, data);
  }
  async filter(data: {
    text: string;
    filters: any[];
    stream?: boolean;
  }): Promise<EventEmitter | any> {
    return await this.fetch(`${this.options.host}/text/filter`, data);
  }
  async google(data: {
    messages: any[];
    model?: string;
    max_tokens?: number;
    temperature?: number;
    id?: string;
  }): Promise<EventEmitter | any> {
    return await this.fetch(`${this.options.host}/text/google`, data);
  }
  async gpt(data: {
    messages: any[];
    model: string;
    max_tokens?: number;
    temperature?: number;
    plugins?: any[];
    id?: string;
  }): Promise<EventEmitter | any> {
    return await this.fetch(`${this.options.host}/text/gpt-new`, data);
  }
  async openchat(data: {
    messages: any[];
    model?: string;
    max_tokens?: number;
    temperature?: number;
    id?: string;
    autoSystemMessage?: boolean;
  }): Promise<EventEmitter | any> {
    return await this.fetch(`${this.options.host}/text/openchat`, data);
  }
  async pawan(data: {
    messages: any[];
    model?: string;
    max_tokens?: number;
    temperature?: number;
    id?: string;
    autoSystemMessage?: boolean;
  }): Promise<EventEmitter | any> {
    return await this.fetch(`${this.options.host}/text/pawan`, data);
  }
}

File: src/modules/text.py
import base
from pyee import EventEmitter

class Text(Base):
    def __init__(self, start):
        super().__init__(start)

    async def alan(self, data):
        response = await self.fetch('text/alan', data)
        return response

    async def anthropic(self, data):
        response = await self.fetch('text/anthropic', data)
        return response

    async def filter(self, data):
        response = await self.fetch('text/filter', data)
        return response

    async def google(self, data):
        response = await self.fetch('text/google', data)
        return response

    async def gpt(self, data):
        response = await self.fetch('text/gpt-new', data)
        return response

    async def huggingface(self, data):
        response = await self.fetch('text/huggingface', data)
        return response

    async def openchat(self, data):
        response = await self.fetch('text/openchat', data)
        return response
